ServantPersistent/src/Models.hs

```diff
@@ -0,0 +1,72 @@
+{-# LANGUAGE DataKinds       #-}
+{-# LANGUAGE TemplateHaskell #-}
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TypeOperators   #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE TypeFamilyDependencies #-}
+{-# LANGUAGE TypeInType #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+module Models
+    where
+
+import Data.Aeson
+import Data.Aeson.TH
+import Database.Persist.Sql
+import Database.Persist.TH
+import GHC.Generics
+import Data.Time.Clock
+import Data.Text
+import Types
+import Control.Monad.IO.Class
+import Control.Monad.Reader
+import Control.Lens
+import Data.ByteString
+import Data.Proxy
+
+share [mkPersist sqlSettings { mpsGenerateLenses = True }, mkMigrate "migrateAll"] [persistLowerCase|
+DBTenant json
```

Comment:
* What does `json` here do?
* Can the Haskell record be called `DBUser` while the table is called `users`?
* Please put a TODO for figuring out how to run custom SQL (triggers and check constraints) when Persistent creates or migrates a table.